pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -846,6 +810,80 @@ func builderSecretVolume(bbs corev1alpha1.BuildBuilderSpec) corev1.Volume {
 	}
 }
 
+func setupBindingVolumesAndMounts(bindings []ServiceBinding) ([]corev1.Volume, []corev1.VolumeMount) {
+	volumes := make([]corev1.Volume, 0)
+	volumeMounts := make([]corev1.VolumeMount, 0)
+
+	for _, binding := range bindings {
+		switch b := binding.(type) {
+		case *corev1alpha1.ServiceBinding:
+			if b.SecretRef != nil {
```

Comment:
Is this necessary?

Reply:
probably not, since the status.binding field needs to be present for the duck type so the secret ref shouldn't be nil

Reply:
i'll take it out

Reply:
actually I think there is a case where the bindings are nil in the provisioned service status so the secret ref would be nil as well